actors/user/src/main/java/org/sunbird/user/actors/JobProfileManagementActor.java

```diff
@@ -0,0 +1,197 @@
+import org.sunbird.learner.util.Util.DbInfo;
+import org.sunbird.user.util.UserActorOperations;
+
+@ActorConfig(
+  tasks = {},
+  asyncTasks = {"upsertUserJobProfile"}
+)
+public class JobProfileManagementActor extends BaseActor {
+
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private Util.DbInfo addrDbInfo = Util.dbInfoMap.get(JsonKey.ADDRESS_DB);
+  private Util.DbInfo jobProDbInfo = Util.dbInfoMap.get(JsonKey.JOB_PROFILE_DB);
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    if (UserActorOperations.UPSERT_USER_JOB_PROFILE
+        .getValue()
+        .equalsIgnoreCase(request.getOperation())) {
+      upsertJobProfileDetails(request);
+    } else {
+      ProjectCommonException exception =
+          new ProjectCommonException(
+              ResponseCode.invalidOperationName.getErrorCode(),
+              ResponseCode.invalidOperationName.getErrorMessage(),
+              ResponseCode.CLIENT_ERROR.getResponseCode());
+      sender().tell(exception, self());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private void upsertJobProfileDetails(Request request) {
+    Map<String, Object> requestMap = request.getRequest();
+    String operationtype = (String) requestMap.get(JsonKey.OPERATION_TYPE);
+    List<Map<String, Object>> reqList =
+        (List<Map<String, Object>>) requestMap.get(JsonKey.JOB_PROFILE);
+    for (int i = 0; i < reqList.size(); i++) {
+      Map<String, Object> jobProfileMap = reqList.get(i);
+      String createdBy = (String) requestMap.get(JsonKey.CREATED_BY);
+      Response addrResponse = null;
+      if (JsonKey.CREATE.equalsIgnoreCase(operationtype)) {
+        jobProfileMap.put(JsonKey.ID, ProjectUtil.getUniqueIdFromTimestamp(i));
+        if (jobProfileMap.containsKey(JsonKey.ADDRESS)) {
+          addrResponse = upsertJobProfileAddressDetails(jobProfileMap, createdBy);
+        }
+        insertJobProfileDetails(requestMap, jobProfileMap, addrResponse, createdBy);
+      } else {
+        if (requestMap.containsKey(JsonKey.IS_DELETED)
+            && null != requestMap.get(JsonKey.IS_DELETED)
+            && ((boolean) requestMap.get(JsonKey.IS_DELETED))
+            && !StringUtils.isBlank((String) requestMap.get(JsonKey.ID))) {
+          deleteJobProfileDetails(requestMap);
+          continue;
+        }
+        if (jobProfileMap.containsKey(JsonKey.ADDRESS)) {
+          addrResponse = upsertJobProfileAddressDetails(jobProfileMap, createdBy);
+        }
+        updateJobProfileDetails(jobProfileMap, addrResponse, createdBy);
+      }
+    }
+    saveUserJobProfileToEs(requestMap);
+  }
+
+  private void saveUserJobProfileToEs(Map<String, Object> requestMap) {
+    Request userRequest = new Request();
+    userRequest.setOperation(UserActorOperations.UPSERT_USER_JOB_PROFILE_TO_ES.getValue());
+    userRequest.getRequest().putAll(requestMap);
+    tellToAnother(userRequest);
+  }
+
+  private void updateJobProfileDetails(
```

Comment:
Create UserJobProfileDao and UserJobProfileDaoImpl with below methods.

insert
update
upsert
delete